ledger/internal/eval.go

```diff
@@ -542,13 +545,17 @@ func (cs *roundCowState) Move(from basics.Address, to basics.Address, amt basics
 		*toRewards = newToRewards
 	}
 
-	toBalNew.MicroAlgos, overflowed = basics.OAddA(toBalNew.MicroAlgos, amt)
-	if overflowed {
-		return fmt.Errorf("balance overflow (account %v, data %+v, was going to receive %v)", to, toBal, amt)
-	}
-	err = cs.putAccount(to, toBalNew)
-	if err != nil {
-		return err
+	// Only deposit if it's meaningful. (Affects balance or rewards state)
```

Comment:
if this is only for sender,s why do we modify the receiver's calculations?

Reply:
There is a small benefit - if someone sends 0 algos to a receiver that has fewer than 1.0 algos, then we can skip it. Today, we would have to write the receiver's account to update the rewardsLevel.

It also feels more consistent.  But I'm open to counter arguments.